app/lib/features/pins/actions/select_pin_attachments.dart

```diff
@@ -0,0 +1,58 @@
+import 'dart:async';
+import 'package:acter/common/models/types.dart';
+import 'package:acter/common/utils/utils.dart';
+import 'package:acter/features/pins/models/create_pin_state/pin_attachment_model.dart';
+import 'package:acter/features/pins/providers/pins_provider.dart';
+import 'package:file_picker/file_picker.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:logging/logging.dart';
+
+final _log = Logger('a3::pins::select::attachment');
+
+FileType attachmentFileType(AttachmentType pinAttachmentType) {
+  switch (pinAttachmentType) {
+    case AttachmentType.image:
+      return FileType.image;
+    case AttachmentType.video:
+      return FileType.video;
+    case AttachmentType.audio:
+      return FileType.audio;
+    case AttachmentType.file:
+      return FileType.custom;
+    default:
+      return FileType.any;
+  }
+}
+
+//Select Attachment
+Future<void> selectAttachment(
+  WidgetRef ref,
+  AttachmentType attachmentType,
+) async {
+  try {
+    final fileType = attachmentFileType(attachmentType);
+    FilePickerResult? result = await FilePicker.platform.pickFiles(
+      type: fileType,
+      allowedExtensions: fileType == FileType.custom ? ['pdf', 'txt'] : null,
```

Comment:
did we limit the extensions before as well? I don't think we want to do that or should do that ...

Reply:
Nope, let me remove this restriction.

Reply:
I thought Image as File make sense in chat but doesn't make sense in Attachment View as there is already image selection available but let me remove it for now as there might be some thoughts arguments.